python/taichi/lang/matrix.py

```diff
@@ -1810,6 +1820,9 @@ def __call__(self, *args):
         return self.cast(Matrix(entries, dt=self.dtype, ndim=self.ndim))
 
     def cast(self, mat):
+        if isinstance(mat, impl.Expr) and mat.ptr.is_tensor():
```

Comment:
1. As it is also a taichi-scope case, is it better to put this after the `in_python_scope` branch?
2. Should `VectorType.cast()` also get modified?